components/ai_chat/core/browser/types.h

```diff
@@ -17,6 +19,19 @@ struct SearchQuerySummary {
   bool operator==(const SearchQuerySummary& other) const = default;
 };
 
+struct Tab {
```

Comment:
I wonder if we could reuse TabData from `brave/brave-core/components/ai_chat/core/common/mojom/tab_tracker.mojom` instead of introducing another `Tab` data type?

Reply:
The main difference is it's origin in my use case, and it's advised in chromium's document to make it clear instead of mixing up origin/url. Would need to introduce optional fields or even union if we want to re-use it, I'm open to it though if that's preferred.

Reply:
Hmmm I'm not 100% sure what's best - I do worry its going to be slightly confusing as there are going to be 3 different `TabData` structs around this feature:
1. https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/webui/tab_search/tab_search.mojom;l=101?q=tab_search%20f:mojom$
2. brave/brave-core/components/ai_chat/core/common/mojom/tab_tracker.mojom
3. This new one

And all three are kind of related to this feature due to where it lives. I maybe have a weak preference for reusing the `TabTracker` one (maybe adding an Origin everywhere) but its only a weak preference, so I'm okay if we leave this as is.

Reply:
Discussed via DM, I ended up with a new one due to 1) origin v.s. url 2) use something from `tab_tracker` which seems to be a specific separate feature seems a bit weird to me.
I think it's possible to change it to something we could share across AI chat component, like moving the struct to a more general file and add fields for different use cases.
Though it's probably not to be blocking this PR, so I'll prefer addressing it in a follow-up if we want to make the change.